libraries/USB/src/USBHID.cpp

```diff
@@ -340,8 +340,8 @@ bool USBHID::SendReport(uint8_t id, const void* data, size_t len, uint32_t timeo
             }
         }
     }
-
     xSemaphoreGive(tinyusb_hid_device_input_mutex);
+    xSemaphoreGive(tinyusb_hid_device_input_sem);
```

Comment:
this semaphore is given in `tud_hid_report_complete_cb`.

Reply:
does it mean I met a situation where `tud_hid_report_complete_cb` is never called?

Reply:
it maybe possible to get into such situation, but in the case you do, a timeout will be reached and you can continue with retry or else.

Reply:
So after a few days of testing I can't reproduce the problem alone, however I found some stability issues that affect all USB devices when USBMSC has an incomplete or failed init. 

I'm not sure if it's a bug or a feature but I can't late-init (on demand) the USBMSC driver without getting side effects.

example 1: comment out the `MSC.begin` in a composite scenario (MSC+Keyboard+Mouse) and try to use the other USB devices.

example 2: use USBMSC with SD.writeRAW and SD.readRAW and boot while no SD card is inserted, then try to use the other USB devices.

In both examples USBSerial eventually crashes and eats the exception message while dying, also HID (mouse or keyboard) freezes the ESP after sending a few reports.

Indeed using this PR does not fix the crash situation, it only makes it happen later.

Should I close this PR as it's obviously unrelated to HID?

Reply:
Yes, you should close this PR. As for MSC, maybe the USB protocol needs to be researched and figure out how to tigger later connect. I suspect that the other devices have issues because the HOST is trying to attach the MSC over and over and is eating the bandwidth. In any  case this is outside of the scope of Arduino as-is. Maybe try in TinyUSB?